cmake/external_dependencies.cmake

```diff
@@ -83,20 +83,29 @@ if(NOT NO_HTTP_CLIENT AND NOT USE_CRT_HTTP_CLIENT)
             include(FindCURL)
             if(NOT CURL_FOUND)
                 message(FATAL_ERROR "Could not find curl")
-            else()
+            endif()
+
+            # When built from source using cmake, curl does not include
+            # CURL_INCLUDE_DIRS or CURL_INCLUDE_DIRS so we need to use
+            # find_package to fix it
+            if ("${CURL_INCLUDE_DIRS}" STREQUAL "" AND "${CURL_LIBRARIES}" STREQUAL "")
+                message(STATUS "Could not find curl include or library path, falling back to find with config.")
+                find_package(CURL)
+                set(CURL_LIBRARIES CURL::libcurl)
+                set(CLIENT_LIBS_ABSTRACT_NAME curl)
+            else ()
                 message(STATUS "  Curl include directory: ${CURL_INCLUDE_DIRS}")
                 message(STATUS "  Curl library: ${CURL_LIBRARIES}")
```

Comment:
suggestion: Is it worth moving this print (or at least `message(STATUS " Curl library: ${CURL_LIBRARIES}")`) after the if so in both configs we print the libraries?

Reply:
i think the sentiment here is good but maybe we need to shuffle the messaging a little bit. 

`Curl library: CURL::libcurl` 

doesnt really make sense to a reader maybe we should change it to a single 

"Curl Target Link: ${CURL_LIBRARIES}"

which is more indicative of what is actually happening in the end. will update it to that, let me know what you think. ideally we should actually change the variable name `CURL_LIBRARIES` to something else but its referenced everywhere and unraveling it might lead to more issues than fixes.

Reply:
Good call 👍 Sounds like a plan